src/chapter_3/PDAs.md

```diff
@@ -0,0 +1,315 @@
+          anchor.getProvider().wallet.publicKey.toBuffer()
+        ],
+        program.programId
+      );
+
+    await program.rpc.createUserStats("brian", {
+      accounts: {
+        user: anchor.getProvider().wallet.publicKey,
+        userStats: userStatsPDA,
+        systemProgram: web3.SystemProgram.programId
+      }
+    });
+
+    expect((await program.account.userStats.fetch(userStatsPDA)).name).to.equal("brian");
+
+    await program.rpc.changeUserName("tom", {
+      accounts: {
+        user: anchor.getProvider().wallet.publicKey,
+        userStats: userStatsPDA
+      }
+    })
+
+    expect((await program.account.userStats.fetch(userStatsPDA)).name).to.equal("tom");
+  });
+});
+```
+
+Exactly as described in the subchapter before this one, we use a `find` function to find the PDA. We can then use it just like a normal address. Well, almost. When we call `createUserStats`, we don't have to add the PDA to the `[signers]` array even though account creation requires a signature. This is because it is impossible to sign the transaction from outside the program. The signature is added when the CPI to the system program is made. We'll cover how that works now.
+
+### Programs as Signers
+
+Creating PDAs requires them to sign the `createAccount` CPI of the system program. How does that work? PDAs are not public keys so it's impossible for them to sign anything. However, PDAs can still pseudo sign CPIs.
+In anchor, to sign with a pda you have to change `CpiContext::new(cpi_program, cpi_accounts)` to `CpiContext::new_with_signer(cpi_program, cpi_accounts, seeds)` where the `seeds` argument are the seeds and the bump the PDA was created with. 
+When the CPI is invoked, for each account in `cpi_accounts` the Solana runtime will check whether`hash(seeds, current_program_id) == account address` is true. If yes, that account's `is_signer` flag will be turned to true.
+This means a PDA derived from some program X, may only be used to sign CPIs that originate from that program X. This means that on a high level, PDA signatures can be considered program signatures.
+
+This is great news because for many programs it is necessary that the program itself takes the authority over some assets.
+For instance, lending protocol programs need to manage deposited collateral and automated market maker programs need to manage the tokens put into their liquidity pools.
+
+Before we conclude this chapter, let's revisit the puppet workspace and add a PDA signature.
+
+First, adjust the puppet-master code:
+```rust,ignore
+use anchor_lang::prelude::*;
+use puppet::cpi::accounts::SetData;
+use puppet::program::Puppet;
+use puppet::{self, Data};
+
+declare_id!("HmbTLCmaGvZhKnn1Zfa1JVnp7vkMV4DYVxPLWBVoN65L");
+
+#[program]
+mod puppet_master {
+    use super::*;
+    pub fn pull_strings(ctx: Context<PullStrings>, bump: u8, data: u64) -> ProgramResult {
+        let cpi_program = ctx.accounts.puppet_program.to_account_info();
```

Comment:
I don't have strong opinions here, but it might be nice to do the idiomatic thing here, which is to pull out the CpiContext creation into a new method on `impl PullStrings` and cpi as 

```
        puppet::cpi::set_data(ctx.accounts.set_data_ctx().with_signer(signer), data)

```